src/Microsoft.DotNet.Interactive/KernelSupportsNugetExtensions.cs

```diff
@@ -241,54 +243,24 @@ internal static KernelCommandInvocation DoNugetRestore()
                     {
                         kernel?.RegisterResolvedPackageReferences(result.ResolvedReferences);
 
+                        install = InstallMessage("Installed packages", (kernel.ResolvedPackageReferences.Select(s => $"{s.PackageName}, {s.PackageVersion}").OrderBy(s => s)));
+                        CreateOrUpdateDisplayValue(context, installingPackagesPropertyName, install, HtmlFormatter.MimeType);
+
                         foreach (var resolvedReference in result.ResolvedReferences)
                         {
-                            if (displayedValues.TryGetValue(
-                                PackageReferenceComparer.GetDisplayValueId(resolvedReference), out var displayedValue))
-                            {
-                                displayedValue.Update(
-                                    $"Installed package {resolvedReference.PackageName} version {resolvedReference.PackageVersion}");
-                            }
-
                             context.Publish(new PackageAdded(resolvedReference, context.Command));
                         }
-
-                        foreach (var package in requestedPackageIds.Values)
-                        {
-                            if (displayedValues.TryGetValue(
-                                PackageReferenceComparer.GetDisplayValueId(package), out var displayedValue))
-                            {
-                                displayedValue.Update(
-                                    $"Installed package {package.PackageName} version {package.PackageVersion}");
-                            }
-                        }
                     }
                     else
                     {
                         var errors = $"{string.Join(Environment.NewLine, result.Errors)}";
-
+                        CreateOrUpdateDisplayValue(context, installingPackagesPropertyName, "", HtmlFormatter.MimeType);
```

Comment:
We probably shouldn't hard-code the MIME type at this layer. If we want more interesting formatting than just text, then pass in a rich object and set up a formatter for it.

Reply:
using a data object + formatter now.